### PR TITLE
all: replace uses of deprecated io/ioutil package

### LINKED ISSUE #709: all: replace ioutil.{Read,Write}File with os.{Read,Write}File
when we drop support of Go-1.15, we should replace all uses of `ioutil.{Read,Write}File` with `os.{Read,Write}File`.

```
$> git grep ReadFile
cmpimg/checkplot.go:69:         got, err := ioutil.ReadFile(path)
cmpimg/checkplot.go:75:         want, err := ioutil.ReadFile(golden)
cmpimg/cmpimg_test.go:21:       got, err := ioutil.ReadFile(filepath.FromSlash("./testdata/failed_input.png"))
cmpimg/cmpimg_test.go:25:       want, err := ioutil.ReadFile(filepath.FromSlash("./testdata/good_golden.png"))
cmpimg/cmpimg_test.go:57:       got, err := ioutil.ReadFile("testdata/approx_got_golden.png")
cmpimg/cmpimg_test.go:72:       got, err := ioutil.ReadFile("testdata/approx_got_golden.png")
cmpimg/cmpimg_test.go:77:       want, err := ioutil.ReadFile("testdata/approx_want_golden.png")
plot_test.go:65:        want, err := ioutil.ReadFile("testdata/legendAlignment_golden.png")
plot_test.go:194:       want, err := ioutil.ReadFile("testdata/glyphbox_golden.png")
vg/fonts/mk-fonts.go:62:        enc, err := ioutil.ReadFile(filepath.Join(fontsDir, "cp1252.map"))
vg/vg_test.go:66:                       want, err := ioutil.ReadFile(name)
vg/vgimg/vgimg_test.go:47:      want, err := ioutil.ReadFile("testdata/issue179_golden.jpg")
vg/vgimg/vgimg_test.go:132:     want, err := ioutil.ReadFile("testdata/issue540_golden.png")
vg/vgimg/vgimg_test.go:137:     got, err := ioutil.ReadFile("testdata/issue540.png")
vg/vgpdf/vgpdf.go:465:          z, err := ioutil.ReadFile(filepath.Join(outdir, "font.z"))
vg/vgpdf/vgpdf.go:472:  j, err := ioutil.ReadFile(filepath.Join(outdir, "font.json"))
vg/vgpdf/vgpdf_test.go:72:                      want, err := ioutil.ReadFile(tc.name)
vg/vgpdf/vgpdf_test.go:134:     want, err := ioutil.ReadFile("testdata/arc_golden.pdf")
vg/vgpdf/vgpdf_test.go:139:     got, err := ioutil.ReadFile("testdata/arc.pdf")
vg/vgpdf/vgpdf_test.go:194:     want, err := ioutil.ReadFile("testdata/issue540_golden.pdf")
vg/vgpdf/vgpdf_test.go:199:     got, err := ioutil.ReadFile("testdata/issue540.pdf")
vg/vgpdf/vgpdf_test.go:246:     raw, err := ioutil.ReadFile("../../plotter/testdata/gopher.png")
vg/vgsvg/vgsvg_test.go:44:      want, err := ioutil.ReadFile("testdata/scatter_golden.svg")
vg/vgsvg/vgsvg_test.go:94:      want, err := ioutil.ReadFile("testdata/scatter_line_golden.svg")

$> git grep WriteFile
plot_test.go:53:                err := ioutil.WriteFile("testdata/legendAlignment_golden.png", buf.Bytes(), 0o644)
plot_test.go:60:        err := ioutil.WriteFile("testdata/legendAlignment.png", buf.Bytes(), 0o644)
plot_test.go:180:       err = ioutil.WriteFile("testdata/glyphbox.png", buf.Bytes(), 0644)
plot_test.go:187:               err := ioutil.WriteFile("testdata/glyphbox_golden.png", buf.Bytes(), 0o644)
vg/fonts/liberation_fonts_generated.go:610:     err = ioutil.WriteFile(_filePath(dir, name), data, info.Mode())
vg/fonts/mk-fonts.go:66:        err = ioutil.WriteFile(filepath.Join(tmpdir, fontsName, "cp1252.map"), enc, 0644)
vg/vg_test.go:78:                               _ = ioutil.WriteFile(got, buf.Bytes(), 0644)
vg/vgimg/vgimg_test.go:57:              _ = ioutil.WriteFile("testdata/issue179.jpg", b.Bytes(), 0644)
vg/vgpdf/vgpdf.go:449:  err = ioutil.WriteFile(fname, font, 0644)
vg/vgpdf/vgpdf.go:454:  err = ioutil.WriteFile(encname, encoding, 0644)
vg/vgpdf/vgpdf_test.go:65:                              err = ioutil.WriteFile(tc.name, buf.Bytes(), 0o644)
vg/vgsvg/vgsvg_test.go:87:              err = ioutil.WriteFile("testdata/scatter_line_golden.svg", b.Bytes(), 0o644)

```

### PR DESCRIPTION
**Pull request by sbinet**
This CL has been mechanically applied through 'eg', to replace all uses
of io/ioutil.{Read,Write}File with os.{Read,Write}File, and similarly
for io/ioutil.Temp{Dir,File}.

Removal of io/ioutil.Discard has been applied manually.

Fixes #709.

(needs #728)

Please take a look.

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -1,15 +1,70 @@
+cloud.google.com/go v0.26.0/go.mod h1:aQUYkXzVsufM+DwF1aE+0xfcU+56JwCaLick0ClmMTw=
```
Comment (kortschak) [inline go.sum:1]: There are a lot of additions to go.sum that look unrelated. Do they go away if you `go mod tidy` or are they coming in from the go-pdf/fpdf?

Comment (sbinet) [reply go.sum]: no, these are coming from a new version of `golang.org/x/exp` (that was collected when this CL was rebased off on a attempt at upgrading Gio).
that new version of `x/exp` has a lot of new OpenTelemetry stuff in it.

I removed that new version of `x/exp` from the `vg/vgpdf` PR.

**Comment by sbinet** [comment]
PTAL

**Review by kortschak — DISMISSED** [review]
